src/map/script.c

```diff
@@ -17172,7 +17172,17 @@ BUILDIN_FUNC(setunitdata)
 			case UMOB_INT: md->status.int_ = (unsigned short)value; status_calc_misc(bl, &md->status, md->level); break;
 			case UMOB_DEX: md->status.dex = (unsigned short)value; status_calc_misc(bl, &md->status, md->level); break;
 			case UMOB_LUK: md->status.luk = (unsigned short)value; status_calc_misc(bl, &md->status, md->level); break;
-			case UMOB_SLAVECPYMSTRMD: md->state.copy_master_mode = value > 0 ? 1 : 0; if (value > 0) { TBL_MOB *md2 = map_id2md(md->master_id); md->status.mode = md2->status.mode; } break;
+			case UMOB_SLAVECPYMSTRMD:
+				md->state.copy_master_mode = value > 0 ? 1 : 0;
```

Comment:
I think this line should be moved below. There's no reason we should set the value for this if it will fail later on if the monster has no master.